### DISCUSSION TITLE
Uploading large file

### DISCUSSION THREAD

**Discussion by knownwhat** [99designs/gqlgen#1781] [Q&A]
### What happened?
Try to upload 15Gb file via gqlgen GraphQL file

### What did you expect?

Correct upload

### Minimal graphql.schema and models to reproduce

mutation function something like this

```
ZipFile(
    Id: ID!
    file: Upload!
  ): ZipArchiveFile
```

### versions
 - `gqlgen version`? 
 - v0.13.0

 - `go version`?
 - go version go1.16.3 darwin/amd64

 - dep or go modules?
only basic

**Comment by knownwhat** [comment]
So, is there some limitations for file uploading in my case and in Gqlgen?

**Reply by lomoval** [reply]
Have you changed `uploadMaxSize` and `uploadMaxMemory` and do you have enough resources (space, memory) to process a file.
I have no problem uploading a 15 GB file (`gqlgen v0.15.1`):
```
$ curl localhost:8087/query  -F operations='{ "query": "mutation ($file: Upload!) { singleUpload(file: $file) { name size } }", "variables": { "file": null } }'  -F map='{ "0": ["variables.file"] }' -F 0=@temp.msi
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15.2G    0    66  100 15.2G      0  50.5M  0:05:08  0:05:08 --:--:--    16{"data":{"singleUpload":{"name":"temp.msi","size":"16370570940"}}}

$ ll temp.msi
-rw-r--r-- 1 User 197121 16370570940 Jan 20 13:58 temp.msi

$ ll ../../Temp/output.bin
-rw-r--r-- 1 User 197121 16370570940 Jan 20 14:05 ../../Temp/output.bin
```
